x/posts/internal/keeper/keeper.go

```diff
@@ -59,12 +59,14 @@ func (k Keeper) SavePost(ctx sdk.Context, post types.Post) {
 	if post.ParentID.Valid() {
 		parentCommentsKey := []byte(types.PostCommentsStorePrefix + post.ParentID.String())
 
-		var commentsIDs []types.PostID
+		var commentsIDs types.PostIDs
 		k.Cdc.MustUnmarshalBinaryBare(store.Get(parentCommentsKey), &commentsIDs)
 
-		commentsIDs = append(commentsIDs, post.PostID)
+		commentsIDs, found := commentsIDs.AppendIfMissing(post.PostID)
```

Comment:
Due to how `AppendIfMissing` it works, I would rename `found` with `appended` for an easier comprehension. 

We can also use one-liner ifs here: 

```suggestion
		if editedIDs, appended := commentsIDs.AppendIfMissing(post.PostID); appended {
			store.Set(parentCommentsKey, k.Cdc.MustMarshalBinaryBare(&editedIDs))
		}
```